### PR TITLE
cleanup: Cleanup content of soft deleted devices.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
In the context of the cleanup script implement cleanup-devices of soft deleted devices:
- cleanup aws s3 bucket update-transaction repo of soft deleted device.
- delete forever update-transaction.
- delete forever update-transaction commit if have no image linked (this mean that image has been cleared but because the commit was linked to update transaction, the commit was not deleted).
- delete forever device. 

FIXES: https://issues.redhat.com/browse/THEEDGE-3489

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by loadtheaccumulator — COMMENTED** [review]
Just the one suggestion

File: cmd/cleanup/cleanupdevices/cleanupdevices.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+// ErrCleanupDevicesNotAvailable error returned when the cleanup devices feature flag is disabled
+var ErrCleanupDevicesNotAvailable = errors.New("cleanup devices is not available")
+
+// ErrDeviceNotCleanUPCandidate error returned when the device is not a cleanup candidate
+var ErrDeviceNotCleanUPCandidate = errors.New("device is not a cleanup candidate")
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go]: done

File: cmd/cleanup/cleanupdevices/cleanupdevices.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+	ImageID         *uint          `json:"image_id"`
+	CommitRepoID    *uint          `json:"commit_repo_id"`
+}
+
+// IsDeviceCandidate to ensure the device is a candidate
+func IsDeviceCandidate(candidateDevice *CandidateDevice) error {
+	if deviceDeletedAtValue, err := candidateDevice.DeviceDeletedAt.Value(); err != nil {
+		return err
+	} else if deviceDeletedAtValue == nil {
+		return ErrDeviceNotCleanUPCandidate
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go]: updated

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Pull request merged**